src/AsyncProducers.cpp

```diff
@@ -494,15 +494,30 @@ class ExpandAcquireNodes : public IRMutator {
     using IRMutator::visit;
 
     Stmt visit(const Block *op) override {
-        Stmt first = mutate(op->first), rest = mutate(op->rest);
-        if (const Acquire *a = first.as<Acquire>()) {
-            // May as well nest the rest stmt inside the acquire
-            // node. It's also blocked on it.
-            return Acquire::make(a->semaphore, a->count,
-                                 mutate(Block::make(a->body, op->rest)));
-        } else {
-            return Block::make(first, rest);
+        vector<Stmt> stmts;
```

Comment:
Here (and elsewhere) it's worth documenting that this structure is done to conserve stack space.

Reply:
Done.